repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java

```diff
@@ -1046,4 +1045,30 @@ public AtlasSearchResult directIndexSearch(SearchParams searchParams) throws Atl
         scrubSearchResults(ret, searchParams.getSuppressLogs());
         return ret;
     }
+
+    private Map<String, Object> getMap(String key, Object value) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(key, value);
+        return map;
+    }
+
+    public List<AtlasEntityHeader> searchUsingTermQualifiedName(int from, int size, String termQName,
+                                                        Set<String> attributes, Set<String>relationAttributes) throws AtlasBaseException {
+        IndexSearchParams indexSearchParams = new IndexSearchParams();
+        Map<String, Object> dsl = getMap("from", from);
+        dsl.put("size", size);
+        dsl.put("query", getMap("term", getMap("__meanings", getMap("value",termQName))));
+
+        indexSearchParams.setDsl(dsl);
+        indexSearchParams.setAttributes(attributes);
+        indexSearchParams.setRelationAttributes(relationAttributes);
+        AtlasSearchResult searchResult = null;
+        try {
```

Comment:
Catch an exception only if you want to handle it in some way
No need for try-catch here